src/main/java/it/gov/pagopa/payments/service/PartnerService.java

```diff
@@ -984,26 +996,39 @@ private PaymentOptionModelResponse getReceiptPaymentOption(
       String noticeNumber,
       String idPa,
       String creditorReferenceId,
+      boolean isStandIn,
       PaymentOptionModel body,
       ReceiptEntity receiptEntity)
       throws FeignException, URISyntaxException, InvalidKeyException, StorageException {
     PaymentOptionModelResponse paymentOption = new PaymentOptionModelResponse();
     try {
-      paymentOption = gpdClient.receiptPaymentOption(idPa, noticeNumber, body);
-      // creates the PAID receipt
-      if (PaymentOptionStatus.PO_PAID.equals(paymentOption.getStatus())) {
+      paymentOption = gpdClient.sendPaymentOptionReceipt(idPa, noticeNumber, body);
+
+      boolean isNotACA = true; // default SERVICE_TYPE_GPD
+      if(paymentOption.getServiceType() != null)
+        isNotACA = !paymentOption.getServiceType().equals(SERVICE_TYPE_ACA);
+
+      // Saving the receipt if the PaymentOption is in the PO_PAID status and service type isn't ACA,
+      // includes all other service types ie GPD, WISP. ACA receipts are saved only if they are stand-in payments.
+      if (PaymentOptionStatus.PO_PAID.equals(paymentOption.getStatus()) && (isNotACA || isStandIn)) {
         this.saveReceipt(receiptEntity);
       }
     } catch (FeignException.Conflict e) {
-      // if PO is already paid on GPD --> checks and in case creates the receipt in PAID status
+      // if PO is already paid on GPD --> checks and in case creates the receipt in PAID status.
       try {
         log.error(
             "[getReceiptPaymentOption] PAA_RECEIPT_DUPLICATA: GPD Conflict Error Response [noticeNumber={}]",
             noticeNumber,
             e);
-        ReceiptEntity receiptEntityToCreate = this.getReceipt(idPa, creditorReferenceId);
-        if (null == receiptEntityToCreate) {
-          // if no receipt found --> save the with PAID receipt
+        boolean receiptNotFoundInStorage = this.getReceipt(idPa, creditorReferenceId) == null;
+
+        boolean isNotACA = true; // default SERVICE_TYPE_GPD
+        if(paymentOption.getServiceType() != null)
+          isNotACA = !paymentOption.getServiceType().equals(SERVICE_TYPE_ACA);
+
+        if (receiptNotFoundInStorage && (isNotACA || isStandIn)) {
+          // if no receipt is not found, or it's a {GPD,WISP} payment, or it's an ACA payments paid in stand-in,
+          // the receipt will be saved in the storage with the PAID status.
```

Comment:
I don't understand: do we want to retry the `saveReceipt` in case of `FeignException.Conflict`?